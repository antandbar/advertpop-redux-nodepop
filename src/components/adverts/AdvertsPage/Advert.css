.advert {
  border-bottom-width: 1px;
  cursor: pointer;
  display: flex;
  padding: 10px 15px 0 15px;
  transition: background-color 0.2s;
}

.advert:hover {
  background-color: rgba(0, 0, 0, 0.03);
}

.advert-content {
  color: rgb(101, 119, 134);
  margin-bottom: 2px;
}

.advert-data {
  color: rgb(20, 23, 26);
  font-weight: bold;
}
