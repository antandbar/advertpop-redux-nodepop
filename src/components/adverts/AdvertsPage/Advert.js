import React from 'react';
import Photo from '../../common/Photo';
import './Advert.css';

const Advert = ({ name, sale, price, tags, photo, isPhoto }) => {
  return (
    <article className="advert bordered">
      {/* Se especifica si se requiere con o sin foto */}
      {isPhoto && (
        <div className="left">
          <Photo className="advert-photo" photo={photo} />
        </div>
      )}

      <div>
        <div className="advert-content">
          <span> El artículo </span>
          <span className="advert-data">{name}</span>
          <span> se </span>
          <span className="advert-data"> {sale ? 'vende' : 'compra'}</span>
          <span> a un precio de </span>
          <span className="advert-data">{price}€</span>
        </div>
        <div className="advert-content">
          <span>Tags asociados: </span>
          <span className="advert-data">
            {/* Se formatena los tags */}
            {JSON.stringify(tags)?.replace(/["'[\]]/g, '')}
          </span>
        </div>
      </div>
    </article>
  );
};

export default Advert;
