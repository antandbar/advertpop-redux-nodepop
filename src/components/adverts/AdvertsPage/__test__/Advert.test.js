import { render } from '@testing-library/react';
import { Provider } from 'react-redux';
import Advert from '../Advert';

describe('AdvertPage', () => {
  test('snapshot', () => {
    const state = {
      adverts: {
        loaded: false,
        data: [],
      },
    };
    const store = {
      getState: () => state,
      dispatch: jest.fn(),
      subscribe: () => {},
    };
    const { container } = render(
      <Provider store={store}>
        <Advert />
      </Provider>,
    );
    expect(container).toMatchSnapshot();
  });
});
