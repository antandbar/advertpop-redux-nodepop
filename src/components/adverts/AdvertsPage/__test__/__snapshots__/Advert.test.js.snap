// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AdvertPage snapshot 1`] = `
<div>
  <article
    class="advert bordered"
  >
    <div>
      <div
        class="advert-content"
      >
        <span>
           El artículo 
        </span>
        <span
          class="advert-data"
        />
        <span>
           se 
        </span>
        <span
          class="advert-data"
        >
           
          compra
        </span>
        <span>
           a un precio de 
        </span>
        <span
          class="advert-data"
        >
          €
        </span>
      </div>
      <div
        class="advert-content"
      >
        <span>
          Tags asociados: 
        </span>
        <span
          class="advert-data"
        />
      </div>
    </div>
  </article>
</div>
`;
